bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,2151 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+1-D radiation model inspired by the PROCESS function "plot_radprofile" in plot_proc.py.
+"""
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+from dataclasses import dataclass, fields
+import matplotlib.pyplot as plt
+import numpy as np
+
+from typing import Dict, Union
+from bluemira.base import constants
+from bluemira.base.constants import ureg
+from bluemira.base.error import BuilderError
+from bluemira.base.parameter_frame import ParameterFrame
+from bluemira.equilibria.equilibrium import Equilibrium
+from bluemira.equilibria.grid import Grid
+from bluemira.equilibria.physics import calc_psi_norm
+from bluemira.display.plotter import plot_coordinates
+from bluemira.geometry.coordinates import Coordinates
+from scipy.interpolate import interp1d
+
+from bluemira.radiation_transport.radiation_tools import (
+    gaussian_decay, 
+    exponential_decay,
+    radiative_loss_function_values,
+    calculate_line_radiation_loss,
+    radiative_loss_function_plot,
+    electron_density_and_temperature_sol_decay,
+    ion_front_distance,
+    random_point_temperature,
+    target_temperature,
+    upstream_temperature,
+    linear_interpolator,
+    interpolated_field_values,
+    
+)
+
+
+if TYPE_CHECKING:
+    from step_reactor.temp_flux_surface_maker import TempFsSolver
+
+
+SEP_CORRECTOR = 5e-3  # [m] DN: 5e-3, SN: 5e-2
+
+
+class Radiation:
+    """
+    Initial and generic class (no distinction between core and SOL)
+    to calculate radiation source within the flux tubes.
+    """
+
+    # fmt: off
+    default_params = [
+        ["n_e_sep", "Electron density at separatrix", 3e+19, "1/m^3", None, "Input"],
+        ["T_e_sep", "Electron temperature at separatrix", 2e-01, "keV", None, "Input"],
+    ]
+    # fmt: on
```

Comment:
All these default_params lists need to go now

```suggestion
    # fmt: off
    default_params = [
        ["n_e_sep", "Electron density at separatrix", 3e+19, "1/m^3", None, "Input"],
        ["T_e_sep", "Electron temperature at separatrix", 2e-01, "keV", None, "Input"],
    ]
    # fmt: on
```

```python
from dataclasses import dataclass

@dataclass
class RadiationParameterFrame(ParameterFrame):
    n_e_sep: Parameter[float] = Parameter(name="n_e_sep", long_name="Electron density at separatrix", value=3e19, unit='1/m^3', source="default")
```

this will convert all the values to SI so be warned.
We might want to not set any defaults with all the data in a config file somewhere but for now the above will work.
For the others you can do eg:

```python
@dataclass
class CoreRadiationParameterFrame(RadiationParameterFrame):
    n_e_0: Parameter[float] =  Parameter(name="n_e_0", long_name="Electron density on axis", value=1.81e+20, unit="1/m^3", source="default")